### PR TITLE
Feature/dashboard event tracking

### PR DESCRIPTION
**Pull request by pattisdr**
:warning: Hold off review until dashboard settles down.
# Purpose

Issue https://openscience.atlassian.net/browse/OSF-5721
Adds high level Google Analytics event tracking to new dashboard to gauge what features are being used.
# Changes

<b>Tracks a combined 96 events in project organizer/dashboard and home page.</b>  All high level tracking, basically clicks, if users filtered, etc.  No information being tracked about specific users.

Event tracking format looks like this: 
ga('send', 'event', [eventCategory], [eventAction], [eventLabel], [eventValue], [fieldsObject]);

Using eventCategory, eventAction, eventLabel.  Can explore sending in eventValue with appropriate fields (must be an integer).

Sample event: 
   `$osf.trackClick('recentActivity', 'filter', 'filter-on-files');`

This indicates that someone within the recent activity widget clicked on the word "files" to filter log actions by files.
# For the reviewer

Clicking on some links trigger the same actions and I've chosen to group them together.  For example, there are two places where you can click to add a component on the dashboard. I've grouped them together. Going through the events and determining if there is a better way to group them would be important.

Also, the vast majority of the events that can be tracked are here, but I am sure I am missing some.  Are there any not included that we need?

### CODE REVIEW

File: website/static/js/pages/home-page.js [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 var $ = require('jquery');
 var m = require('mithril');
+var $osf = require('js/osfHelpers');
```
Comment (samanehsan) [inline website/static/js/pages/home-page.js]: This file already requires `js/osfHelpers` a few lines up, so this can be removed.

File: api/logs/permissions.py [hunk 2]
```diff
@@ -12,8 +12,14 @@ class ContributorOrPublicForLogs(permissions.BasePermission):
     def has_object_permission(self, request, view, obj):
         assert isinstance(obj, (NodeLog)), 'obj must be a NodeLog, got {}'.format(obj)
 
-        for node_id in obj._backrefs['logged']['node']['logs']:
-            node = get_object_or_error(Node, node_id, display_name='node')
-            if ContributorOrPublic().has_object_permission(request, view, node):
+        if obj._backrefs.get('logged'):
```
Comment (samanehsan) [inline api/logs/permissions.py:15]: Was this backrefs change meant to be a part of a different PR?

Comment (samanehsan) [inline api/logs/permissions.py:15]: Oh, it looks like it's from a commit that was merged into `caneruguz/dashboard-redesign-2` Can you please bring this up to date with that branch @pattisdr?

Comment (pattisdr) [inline api/logs/permissions.py:15]: Unless @caneruguz has told you differently, I was told to hold off on event tracking for a few days(as the dashboard itself is changing very rapidly right now).  Caner, @hmoco, and I are responding to Steve's review right now.  When the dashboard is more settled, then I will bring this up to date.  I think the review should wait until then.

Comment (pattisdr) [inline api/logs/permissions.py:15]: For example, it looks like the recent activity widget was removed from the home page for now, so all of my event tracking of that module needs to go.  But I will wait until I see what else is changing.

Comment (samanehsan) [inline api/logs/permissions.py:15]: Oh ok, that makes sense! Thanks for letting me know!

Comment (pattisdr) [inline api/logs/permissions.py:15]: Yeah, I should have put a note on this!

**Comment by pattisdr** [comment]
Going to make this a PR to feature/dashboard instead now that dashboard-redesign-2 is merged.
